src/logmessage.cpp

```diff
@@ -108,7 +115,19 @@ namespace g3 {
       return out;
    }
 
-
+   std::string LogMessage::message() const {
+      std::string msg;
+      if (!_wmessage.empty())
+      {
+#if !defined(__GNUC__) || (__GNUC__ >= 5)
+         std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> convert;
+         msg.append(_message.append(convert.to_bytes(_wmessage)));
```

Comment:
Yes, at least gcc 4.7 does not know the codecvt header and has no wstring_convert. Current gcc does, though: [ideone example](http://ideone.com/ft0w9k)
I am not sure which stdlibc++ is needed for this to work. Here I just assume gcc 5 or above is ok. The proper way to do this is to move decision to the cmake file and include a compilation test and then have a macro WSTRING_CONVERT_AVAILABLE that decides here. 
Should I do it that way?

Also I should mention that what I do here is EXTREMELY inefficient. The bulk of the time is spent constructing the convert object. This is meant as a fall-back when no custom sink is used. wstring_convert [is not thread safe](http://stackoverflow.com/questions/16644257/are-member-functions-of-stdwstring-convert-thread-safe), so using a static wstring_convert would entail synchronization or the guarantee that the message() function is only ever called from one thread or a thread local wstring_convert object. 
What is your feeling? How important is performance on the log worker side? I would assume that whoever uses wstring logging has a custom sink.